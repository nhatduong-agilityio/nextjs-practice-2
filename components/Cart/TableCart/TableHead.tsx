// Libs
import { Thead, Tr } from '@chakra-ui/react'
import React from 'react'

// Components
import TableColumn from '@components/Cart/TableCart/TableColumn'

const TableHead = () => (
  <Thead>
    <Tr w='full'>
      <TableColumn width='40%' label='Product' />
      <TableColumn width='15%' label='Quantity' />
      <TableColumn width='25%' label='Color' />
      <TableColumn width='15%' label='Price' />
      <TableColumn width='5%' label='' />
    </Tr>
  </Thead>
)

export default TableHead
