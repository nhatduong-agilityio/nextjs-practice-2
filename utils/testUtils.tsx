import { ChakraProvider } from '@chakra-ui/react'
import { AuthProvider } from '@store/providers/AuthProvider'
import { LoadingProvider } from '@store/providers/LoadingProvider'
import type { RenderOptions, RenderResult } from '@testing-library/react'
import { render } from '@testing-library/react'
import { customTheme } from '@themes/index'
import type { ReactElement } from 'react'

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <ChakraProvider theme={customTheme}>
      <LoadingProvider>
        <AuthProvider>{children}</AuthProvider>
      </LoadingProvider>
    </ChakraProvider>
  )
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>): RenderResult =>
  render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
