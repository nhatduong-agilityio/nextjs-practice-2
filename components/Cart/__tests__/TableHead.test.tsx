// Utils
import { render } from '@utils/testUtils'

// Components
import { Table } from '@chakra-ui/react'
import TableHead from '../TableCart/TableHead'

describe('TableHead render', () => {
  it('Should TableHead match Snapshot', () => {
    const { container } = render(
      <Table>
        <TableHead />
      </Table>,
    )

    expect(container).toMatchSnapshot()
  })
})
