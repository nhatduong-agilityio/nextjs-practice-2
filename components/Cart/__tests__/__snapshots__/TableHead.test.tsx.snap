// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHead render Should TableHead match Snapshot 1`] = `
<div>
  <table
    class="chakra-table css-5605sr"
  >
    <thead
      class="css-0"
    >
      <tr
        class="css-1ki54i"
      >
        <th
          class="css-1lbjhhx"
        >
          <p
            class="chakra-text css-1x05orc"
            title="columnLabel"
          >
            Product
          </p>
        </th>
        <th
          class="css-j025zv"
        >
          <p
            class="chakra-text css-1x05orc"
            title="columnLabel"
          >
            Quantity
          </p>
        </th>
        <th
          class="css-151n1ag"
        >
          <p
            class="chakra-text css-1x05orc"
            title="columnLabel"
          >
            Color
          </p>
        </th>
        <th
          class="css-j025zv"
        >
          <p
            class="chakra-text css-1x05orc"
            title="columnLabel"
          >
            Price
          </p>
        </th>
        <th
          class="css-inag1h"
        >
          <p
            class="chakra-text css-1x05orc"
            title="columnLabel"
          />
        </th>
      </tr>
    </thead>
  </table>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
